src/riak_kv_coverage_filter.erl

```diff
@@ -150,26 +154,26 @@ build_item_filter(FilterInput) ->
 
 
 %% @private
-build_preflist_fun(Bucket, CHBin) ->
+build_preflist_fun(Bucket, CHBin, KeyConvFn) ->
     fun({o, Key, _Value}) -> %% $ index return_body
-            ChashKey = riak_core_util:chash_key({Bucket, Key}),
+            ChashKey = riak_core_util:chash_key({Bucket, KeyConvFn(Key)}),
```

Comment:
Theoretically, you could obtain a DDL from `Bucket` and call `riak_ql_ddl:get_partition_key` on the spot here, without the need to create that function in riak_kv_ts_svc:sub_tslistkeysreq` and tread it all the way here. This may or may not cause any performance overhead, but it would make the diff a lot smaller. I'm not urging you to do it this way, though; just, what were your arguments for the design decision you have taken?